### PR TITLE
[layout/mediaQuery] initial breakpoints

### PR DESCRIPTION
**Pull request by carlosvillu**

### CODE REVIEW

**Review by jcger — DISMISSED** [review]

File: components/layout/mediaQuery/src/index.js [hunk 1]
```diff
@@ -10,9 +10,19 @@ import {BREAKPOINTS} from './breakpoints'
 
 const LayoutMediaQueryFactory = BREAKPOINTS =>
   class extends Component {
+    static defaultProps = {
+      initialMediaQueries: {}
+    }
+
+    static propTypes = {
+      /** MediaQueries to be used in the first render. Until of be called componentDidMount and setup the proper MQ´s Usefull for SSR */
```
Comment (midudev) [inline components/layout/mediaQuery/src/index.js]:
```
  /**
   * MediaQueries to be used in the first render until componentDidMount is called and it setups the proper media query. Useful for SSR for avoiding re-renders.
   */
```

Comment (midudev) [reply components/layout/mediaQuery/src/index.js]: There're some typos fixed and used the correct format for the comment.

File: demo/layout/mediaQuery/playground [hunk 2]
```diff
@@ -53,7 +53,7 @@ return (
     <p>Resize the width of the viewport or the container with dashed blue border to see the behavior</p>
     <h2>Basic Usage (default values)</h2>
     <div style={stylesSection}>
-      <LayoutMediaQuery>
+      <LayoutMediaQuery initialMediaQueries={{XS: true, S: true}}>
```
Comment (midudev) [inline demo/layout/mediaQuery/playground:56]: How is that working with two initialMediaQueries as true?

**Review by midudev — COMMENTED** [review]
I miss further documentation in order to understand how it works and how to use it.

File: components/layout/mediaQuery/src/index.js [hunk 3]
```diff
@@ -57,10 +68,19 @@ const LayoutMediaQueryFactory = BREAKPOINTS =>
       if (!shallowEqual(result, params)) this.setState({params: result})
     }
 
+    shouldComponentUpdate(_, nextState) {
+      return !shallowEqual(this.state.params, nextState.params)
```
Comment (carlesnunez) [inline components/layout/mediaQuery/src/index.js:74]: This hook is not required as long while you are using a PureComponent.
https://reactjs.org/docs/shallow-compare.html

In any case.. im not pretty sure about that we need a pure component on this piece. Why did you changed the stateless version for a pure component?

Comment (carlosvillu) [reply components/layout/mediaQuery/src/index.js:74]: I had remove the pure component. And left the shouldComponentUpdate. The PureComponent only didnt works.

**Pull request closed without merging**